### ISSUE TITLE
Rule: lit/lifecycle-super should not fail in declarations

### ISSUE THREAD

**Issue by MRyves** [43081j/eslint-plugin-lit#224]
In our project we use a combination of js/ts and some lit-components are still written in vanilla js.
Let's say you have following lit component:

```javascript
// File: Foo.js
import { LitElement, html, css } from 'lit';

export class Foo extends LitElement {
  
  connectedCallback() {
    super.connectedCallback();
  }

  disconnectedCallback() {
    super.disconnectedCallback();
  }

  render() {
    return html`<p>Hello from Foo!</p>`;
  }
}

customElements.define('my-foo', Foo);
```
Now here is the class declaration:
```typescript
// File: Foo.d.ts
import type { LitElement, TemplateResult } from 'lit';

export class Foo extends LitElement {
  connectedCallback(): void;
  disconnectedCallback(): void;
  render(): TemplateResult;
}
```

Running eslint with the rule ` lit/lifecycle-super` set to throw error if it fails we get following output:
```
  4:3  error  You must call `super.connectedCallback` to avoid interrupting the lit rendering lifecycle     lit/lifecycle-super
  5:3  error  You must call `super.disconnectedCallback` to avoid interrupting the lit rendering lifecycle  lit/lifecycle-super
```

I guess this should not be the case as it is simply the declration of a class and not the actual implementation.

**Comment by 43081j** [comment]
im a little lost

i think because your error doesn't specify what file you're referring to. are you saying it is linting your `d.ts`?

you should probably tell it not to, if that's the case. that's out of our control as we just get given the AST
